ec/ecresource/extensionresource/read_test.go

```diff
@@ -0,0 +1,143 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package extensionresource
+
+import (
+	"context"
+	"testing"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/api/mock"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/terraform-provider-ec/ec/internal/util"
+)
+
+func Test_readResource(t *testing.T) {
+	tc200 := util.NewResourceData(t, util.ResDataParams{
+		ID:     "12345678",
+		State:  newExtension(),
+		Schema: newSchema(),
+	})
+	wantTC200 := util.NewResourceData(t, util.ResDataParams{
+		ID:     "12345678",
+		State:  newExtension(),
+		Schema: newSchema(),
+	})
+
+	tc500Err := util.NewResourceData(t, util.ResDataParams{
+		ID:     "12345678",
+		State:  newExtension(),
+		Schema: newSchema(),
+	})
+	wantTC500 := util.NewResourceData(t, util.ResDataParams{
+		ID:     "12345678",
+		State:  newExtension(),
+		Schema: newSchema(),
+	})
+
+	tc404Err := util.NewResourceData(t, util.ResDataParams{
+		ID:     "12345678",
+		State:  newExtension(),
+		Schema: newSchema(),
+	})
+	wantTC404 := util.NewResourceData(t, util.ResDataParams{
+		ID:     mock.ValidClusterID,
+		State:  newExtension(),
+		Schema: newSchema(),
+	})
+	wantTC404.SetId("")
+
+	type args struct {
+		ctx  context.Context
+		d    *schema.ResourceData
+		meta interface{}
+	}
+	tests := []struct {
+		name   string
+		args   args
+		want   diag.Diagnostics
+		wantRD *schema.ResourceData
+	}{
+		{
+			name: "returns nil when it receives a 200",
+			args: args{
+				d: tc200,
+				meta: api.NewMock(mock.New200StructResponse(models.Extension{
+					Name:          stringPtr("my_extension"),
+					ExtensionType: stringPtr("bundle"),
+					Description:   "my description",
+					Version:       stringPtr("*"),
```

Comment:
There's an `ec.String` provided by the `elastic/cloud-sdk-go/pkg/util/ec` that can be used here.

Reply:
fixed by dfbcd34